### PR TITLE
Notifying users that new releases are not planned.

### PR DESCRIPTION
**Pull request by esplinr**
As our focus moves from the Indy SDK to replacement libraries, we should set the expectation with users of Indy SDK that future releases are not planned. We should also encourage contributors to participate in replacement efforts rather then submitting PRs here that may never be reviewed.

Signed-off-by: Richard Esplin <richard-oss@esplins.org>

### CODE REVIEW

**Review by jovfer — APPROVED** [review]

**Review by dhh1128 — APPROVED** [review]

**Review by brentzundel — APPROVED** [review]

**Comment by esplinr** [comment]
Note that this PR should not be merged until we have the chance to discuss it in the next Indy Contributors call.

**Review by m00sey — COMMENTED** [review]
I would rebase to make it a single comment.

On the `esplinr/master` branch:

`git rebase HEAD~2`
`git push --force-with-lease`

**Comment by esplinr** [comment]
I made the change that @m00sey suggested.

**Review by swcurran — CHANGES_REQUESTED** [review]
Others are wanting to do releases.

**Comment by esplinr** [comment]
I believe that I've addressed all concerns that have been raised and this is ready for merging.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+An alternative approach to the Indy SDK is being developed which divides LibIndy into multiple shared libraries which will be part of the Indy and Aries projects:
+
+* a verifiable data registry for communication with Indy ledgers,
+* an encrypted storage service for Hyperledger Aries agents,
+* a credential exchange library,
+* and a library of shared Rust functions.
+
+Progress is recorded in the notes for the [Indy Contributors Meeting](https://wiki.hyperledger.org/display/indy/Indy+Contributors+Meeting).
+
+[Indy Node](https://github.com/hyperledger/indy-node) will continue to be maintained and improved.
```
Comment (swcurran) [inline README.md:13]: Please reword to:

'Hyperledger Indy's ledger implementation, [Indy Node](https://github.com/hyperledger/indy-node) and related codebases, continue to be maintained and improved.'

I'd suggest moving this up to right below the line about "New Indy-SDK releases..."

File: README.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+
+If you would like to continue the development of Indy SDK by implementing a feature, bug fix, or publishing a release, please contact us and the appropriate guidance and controls can be granted. You can find us in [#indy-contributors on Hyperledger Rocket Chat](http://chat.hyperledger.org/channel/indy-contributors) or [the Indy mailing list](http://lists.hyperledger.org/g/indy/).
```
Comment (swcurran) [inline README.md:15]: Suggest changing the "us" to "the maintainers" in the first sentence.

**Review by swcurran — CHANGES_REQUESTED** [review]
I've made some adjustments to the wording that I think are more appropriate and reflective of the situation with this repo, with an emphasis that it is just this repo that will be replaced over time.

**Comment by Patrik-Stas** [comment]
Can we find way around this? I see many useful pending PRs (especially asynchronicity in libindy 🚀 ) and I don't think the whole repo needs to "die" like this. Even though BC gov is doing great job providing alternatives in form of individual smaller libraries, libindy still has its users.
The major issue I see with this repo is the pipeline. It's difficult for both maintainers and contributors - pipeline is unstable and private (no way for contributor to see Jenkins logs). Contributors don't know why their PR are failing so it's impossible to iterate, for maintainers the maintenance itself is expensive because managing Jenkins infra and pipelines is quite resource-expensive in my view.

My suggestion:
1. Create Github Actions based pipeline with limited scope - obviously we can't do everything Jenkins was trying to do using GA runner infrastructure (test on multiple linux distributions for example). Smaller but *working and fast* CI is better than broken giant one. I think the starting scope for new CI should be compiling and test libindy and its wrappers (libvcx excluded from CI completely). Then start adding publishing or other stuff. Contributing GA CI code is much more accessible than contributing to Jenkins CI in its current form, so I believe it might organically grow if there will be interest in libindy. 
2. If the proposal seems reasonable way to keep indysdk breathing, we are open to contribute GA actions (in fact, I've started digging into this about a year ago already https://github.com/hyperledger/aries-vcx/actions/runs/438470126 ) as we've gained quite rich experience with them implementing GA for `libvcx` fork `aries-vcx` - see example of our CI run 
https://github.com/hyperledger/aries-vcx/runs/1595971044

**Comment by swcurran** [comment]
I agree @Patrik-Stas that this not be done or be done carefully -- that's what my suggested changes are about.

The GA changes are welcome, but in our opinion, the indy-node (and related repos) are more crucial at this time. If you and your team have GA experience, we would love your help on that.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,22 @@
+# Project Status
+Releases are currently only created as they are needed. We are actively seeking contributors to take ownership of creating new releases.
```
[the remark this answers is not in the document]
Comment (esplinr) [reply README.md:2]: The suggested wording implies that we will work on releases if people ask us to, so I changed it a bit to preserve the sense that we are looking for people who want to help with releases.

File: README.md (hunk 2 above)
Comment (esplinr) [reply README.md:15]: I left it ambiguous because it is unclear who "the maintainers" are.

**Comment by esplinr** [comment]
I believe I have now incorporated all feedback so far.

**Comment by esplinr** [comment]
@Patrik-Stas That is a generous offer, and resolves one of the biggest concerns with the Indy SDK. I welcome your team's contribution!

However, I still think a message like this is needed. By making it clear that the Indy SDK needs new maintainers, people will know that they either need to pitch in, or move on.

There are many useful PRs, but no one to review them. The list of unresolved issues in this repo is growing, and doesn't include the many issues in Jira that were thrown away. Many of the wrappers are very out-of-date. And there are reports that it doesn't build on the latest Ubuntu.

Your effort to migrate Indy SDK to GitHub Actions will significantly lower the cost of new contributions. I hope that others choose to join you and that it breathes new life into the project. If you decide that you want to take full ownership over the repo, I'll close this PR and encourage my team to keep contributing as well.

**Comment by fedyiv** [comment]
Hello, can you please advise what is the libindy replacement?

**Comment by swcurran** [comment]
There are three components that have been created as an alternative to the indy-sdk:

- indy-vdr enables integration with the ledger. It can be run embedded in an app or as a service with an API
  - https://github.com/hyperledger/indy-vdr
- indy-shared-rs implements the crypto (including AnonCreds) for indy by wrapping Ursa
  - https://github.com/hyperledger/indy-shared-rs
- the storage in indy (unfortunately called "indy-wallet") can be replaced by aries-askar within Aries frameworks
  - https://github.com/hyperledger/aries-askar
  
These components are complete but have not been used in other than test scenarios. Once the next version of ACA-Py is released, we'll be adding support for these into main so that they can be used in production use cases.
  
 Also being added in different Aries frameworks are ways to use different ledgers and different verifiable credential formats.
 
 As noted in comments above, ABSA and others is likely to continue to evolve the indy-sdk and to continue to release it.

**Review by m00sey — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,3 +1,21 @@
+# Indy SDK Status
+The Indy community is transitioning to a new approach to Indy client applications, and so new releases of the Indy SDK are currently only created as they are needed. We are actively seeking contributors to take ownership of creating new releases. If you are interested in helping with a new release of the Indy-SDK,please reach out using the contact information below.
```
Comment (swcurran) [inline README.md:2]: Typo -- missing space after ","

**Review by swcurran — CHANGES_REQUESTED** [review]
LGTM other than the one comment about the typo.

File: README.md (hunk 4 above)
Comment (esplinr) [reply README.md:2]: Nice catch!

**Review by swcurran — APPROVED** [review]

**Comment by swcurran** [comment]
@ianco @Patrik-Stas  -- do either of you know how to get the pre-merge checks to pass?

**Comment by ianco** [comment]
(ci) please test

**Pull request closed without merging**